### DISCUSSION TITLE
已存在的分区能否拆分成多个分区

### DISCUSSION THREAD

**Discussion by cdavis26424** [apache/doris#10515] [A - General / Q&A] [answered]
版本 0.15

原来手动创建了时间范围的分区 2022-01-01 至 2022-07-01
现在希望把分区设置成一个月一个分区
使用如下命令
ALTER TABLE test_db.test_tab SET
(
    "dynamic_partition.enable" = "true",
    "dynamic_partition.time_unit" = "MONTH",
    "dynamic_partition.start" = "-2147483648",
    "dynamic_partition.end" = "3",
    "dynamic_partition.prefix" = "p",
    "dynamic_partition.buckets" = "32",
    "dynamic_partition.replication_num" = "3",
    "dynamic_partition.create_history_partition" = "true",
    "dynamic_partition.history_partition_num" = "7"
)
创建了未来的分区，但是没有创建过去的分区，而原来 2022-01-01 至 2022-07-01的分区依然存在。
希望达成的目标：保留数据，移除 2022-01-01 至 2022-07-01分区，生成一月至六月 每月一个分区。
能否实现？如何实现？

**Comment by hf200012** [comment, marked as answer]
不可以
你可以先创建一个月的范围临时分区，然后将这将这一个月的数据全部insert到这个临时分区里，成功删除这些天分区，最后讲临时分区重命名成正式分区，请参照：https://doris.apache.org/zh-CN/docs/advanced/partition/table-tmp-partition.html#%E8%A7%84%E5%88%99
